clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -94,6 +96,36 @@ def SInt16 : SInt<16>;
 def SInt32 : SInt<32>;
 def SInt64 : SInt<64>;
 
+//===----------------------------------------------------------------------===//
+// FloatType
+//===----------------------------------------------------------------------===//
+
+class CIR_FloatType<string name, string mnemonic> 
+    : CIR_Type<name, mnemonic,
+               [DeclareTypeInterfaceMethods<DataLayoutTypeInterface>],
+               "::mlir::cir::FloatType"> {}
+
+def CIR_Single : CIR_FloatType<"Single", "float"> {
+  let summary = "CIR single-precision float type";
+  let description = [{
+    Floating-point type that represents the `float` type in C/C++. Its
+    underlying floating-point format is the IEEE-754 binary32 format.
+  }];
+}
+
+def CIR_Double : CIR_FloatType<"Double", "double"> {
+  let summary = "CIR double-precision float type";
+  let description = [{
+    Floating-point type that represents the `double` type in C/C++. Its
+    underlying floating-point format is the IEEE-754 binar64 format.
+  }];
+}
+
+// Constraints
+
+def CIR_AnyFloat: Type<
+    CPred<"$_self.isa<::mlir::FloatType, ::mlir::cir::FloatType>()">>;
```

Comment:
You're working around what @sitio-couto suggested. You still have only one floating point type. Each Op should be a different floating point type, and an interface could be created to share common methods and whatnots. In other words, you don't need a `mlir::cir::FloatType`. With that in mind, this would become:
```
// `::mlir::cir::FloatType` below is f32, not the initial `::mlir::cir::FloatType` from this PR.
def CIR_AnyFP: Type<
    CPred<"$_self.isa<::mlir::cir::FloatType, mlir::cir::DoubleType>()">>;
```

Reply:
Updated. I have replaced the `mlir::cir::FloatType` with an interface `FPTypeInterface`.